### ISSUE TITLE
Add aria only SVG nodes to improve accessibility

### ISSUE THREAD

**Issue by domoritz** [vega/vega#2552]
Screenreaders read accessibility elements in the order that they appear in the DOM. Since SVG does not have a way to change the index of elements and legends have to appear on top of marks, [legends appear after the marks in screenreaders](https://github.com/vega/vega/issues/2523). Ideally, we would provide the context of the legend first, though.

Instead of relying on a z-index, we could instead insert an invisible element in the SVG that has only aria properties for a legend/title defined. The invisible element can be inserted at any point. Then, we can hide the legend/title with `aria-hidden: true`.

**Comment by mgifford** [comment]
Worth checking out Carie Fisher's work on SVGs https://www.smashingmagazine.com/2020/03/accessible-svgs-inclusiveness-beyond-patterns/

There are lots of ways to make SVG's more accessible. ARIA is powerful, but when used incorrectly actually reduces accessibility.  ARIA should be like salt and used in moderation.
